extensions/oplogPopulator/modules/Connector.js

```diff
@@ -109,13 +121,14 @@ class Connector {
      * Add bucket to this connector
      * Connector is updated with the new bucket list
      * @param {string} bucket bucket to add
-     * @param {boolean} [doUpdate=true] updates connector if true
+     * @param {boolean} [doUpdate=false] updates connector if true
      * @returns {Promise|undefined} undefined
      * @throws {InternalError}
      */
-    async addBucket(bucket, doUpdate = true) {
+    async addBucket(bucket, doUpdate = false) {
+        this._buckets.add(bucket);
+        this._updateConnectorState(true);
```

Comment:
What happen if the `updatePipeline` fails? do we still add the bucket to the list, as it is done before the try/catch?
(I'm curious to know if the approach will be stable in case of errors, so we don't end up in an inconsistent state, but I may lack a bit of context on this one)

Reply:
Yes the bucket is still added to the list but `_state.bucketsGotModified` stays as true, which will trigger an update of the connector, so we will basically be retrying until it succeeds. We also get an error log each time it fails

Reply:
i added this https://github.com/scality/backbeat/compare/17128feaebde3bc87cd6fe8afe642ba5063f0702..a0941c5e9287ea76513da7b7e7104476a3c38bc9 just to be sure the retry gets triggered